src/components/Heading/Heading.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+export type HeadingLevelType = "h2" | "h3" | "h4" | "h5";
+
+type HeadingProps = {
+  children: React.ReactNode;
+  level?: HeadingLevelType;
+  className?: string;
+};
+
+const Heading: React.FunctionComponent<HeadingProps> = ({
+  children,
+  level = "h2",
+  className
+}) => {
+  const HeadingTag = {
+    h2: "h2",
+    h3: "h3",
+    h4: "h4",
+    h5: "h5"
+  }[level] as keyof JSX.IntrinsicElements;
```

Comment:
```suggestion
  const HeadingTag = level as keyof JSX.IntrinsicElements;
```

Please reconsider the structure above. It takes a string literal and uses an object with the same keys to map to strings with the same values. All in all the allowed values are repeated three times. That should not be necessary. The suggestion above tries to fix that. Perhaps try that out or consider other options.